OpenRA.Mods.Sp/Traits/World/CloudSpawner.cs

```diff
@@ -95,5 +104,61 @@ void SpawnCloud(Actor self)
 
 			self.World.AddFrameEndTask(w => w.Add(new Cloud(self.World, animation, startEdge, finishEdge, facing, info)));
 		}
+
+		void PreSpawnClouds(World world)
+		{
+			var facing = 256 * info.WindDirection / 32;
+			var direction = WAngle.FromFacing(facing);
+			var delta = new WVec(0, -1024, 0).Rotate(WRot.FromFacing(facing));
+
+			/*
+			 * The folloing codes find the middle point in map. The paths of cloud which cover
+			 * this point will always be the longest path.
+			 *
+			 * By using this longest path, we can figure out how many clouds should
+			 * spawn when a cloud go over a completely longest path, which should be
+			 * neither too little nor too many compared with the clouds per map-cell later.
+			 *
+			 */
+			var lPosition = new MPos(world.Map.MapSize.X / 2, world.Map.MapSize.Y / 2);
```

Comment:
What are these l prefixes?

Reply:
Longest.
Which means the longest path for a cloud can go, under the given wind direction.

Reply:
Shall I use the full prefix "longest"?

Reply:
OpenRA code convention is for self-explaining code, which means it's against abbreviations. This is a great example when your abbreviations caused the need of the huge comment block in front of these operations.

Reply:
Fixed